compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -946,14 +948,34 @@ impl<'local, 'interner> Interpreter<'local, 'interner> {
     }
 
     /// Given the result of a `cmp` operation, convert it into the boolean result of the given operator.
-    fn evaluate_ordering(&self, ordering: Value, operator: BinaryOpKind) -> IResult<Value> {
-        let ordering = match ordering {
-            Value::Struct(fields, _) => match &*fields.into_iter().next().unwrap().1.borrow() {
-                Value::Field(ordering) => *ordering,
-                _ => unreachable!("`cmp` should always return an Ordering value"),
-            },
-            _ => unreachable!("`cmp` should always return an Ordering value"),
+    fn evaluate_ordering(
+        &self,
+        ordering: Value,
+        operator: BinaryOpKind,
+        location: Location,
+    ) -> IResult<Value> {
+        let field_ordering = match &ordering {
+            Value::Struct(fields, _) => {
+                let first_field = fields.iter().next();
+                match first_field {
+                    Some((_, value)) => match &*value.borrow() {
+                        Value::Field(ordering) => Some(*ordering),
```

Comment:
Technically this still accepts any `struct` where the first field is a `Field`. Should we check that the `Type` in the `Struct` bears the expected `name` for example?

Reply:
For built-in functions we rely on getting the type via the type signature of the called function. We can't rely on the types from impls but I wonder if there's a way we could access the original trait and its return type here.

Reply:
I added a check on Ordering name.